src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -207,13 +211,14 @@ public virtual async Task<T> GetConfigurationAsync(CancellationToken cancel)
                 finally
                 {
                     _configurationNullLock.Release();
+                    Interlocked.Exchange(ref _configurationRetrieverState, ConfigurationRetrieverIdle);
```

Comment:
I'm not convinced this needs to be an Interlocked.Exchange either. A write to an int is atomic, and we don't need to write and also know the original value in one operation.